### ISSUE TITLE
Option for voyage recommendation calculations to be balanced or weighted

### ISSUE THREAD

**Issue by SodPen1** [IAmPicard/StarTrekTimelinesSpreadsheet#195]
**Is your feature request related to a problem? Please describe.**
Not related to a problem.  

**Describe the solution you'd like**
Depending on the intent and goal of individual voyages, sometimes it's better to go with a balanced approach and sometimes it's better to go with a weighted approach when deciding upon crew composition.  

Ideally, what I'd like to see in the application is a checkbox option included alongside the "include active crew" option for "calculate weighted."  When left unchecked, the application would calculate the best crew and recommend exactly as it does now, which is the balanced approach (all six disciplines given equal weight).  When checked, the application would calculate the balanced recommendations, then determine the weakest of the four non-priority disciplines, then sacrifice the crew in those two slots, replacing them with the two crew that would bolster the primary discipline, then recommend a 12-crew compliment based on that composition.  For extra tweaking, subset weighted options for bolstering primary discipline, secondary discipline, or both could be included. 

**Describe alternatives you've considered**
At present, when I want to run a weighted voyage, I have to take the balanced recommendation and then manually drop and replace the crew from the weakest discipline and launch the voyage manually within the game.  I've considered continuing to do that, and decided that it's a quality of life feature that I'd really like to see in the application natively.

**Additional context**
N/A

**Comment by SodPen1** [comment]
You are, of course, absolutely correct in the context of wanting to run optimized voyages with the goal being the longest voyage time possible.  However, and this is where I must apologize for being remiss in my original post and wording, there are several mitigating circumstances where this is not the most desirable outcome or goal.  Specifically, when you factor in the differences between players who are playing pure f2p and those who spend various amounts of money on dilithium to use with voyages.  Once real-world money becomes a factor, then it can drastically alter just how long you want your voyages to run depending on the investment / return for voyages of various lengths.  Often times, you would rather (by way of example) run two 12-hour voyages instead of a single 24.  Or vice versa.  Granted, in the upper ranges of 12+ hour voyages, the difference between balanced and weighted approach is negligible at best, mathematically, but for players who prefer to run voyages in the 4-12 hour range either because they're newer players and just don't have the crew to run longer times natively, or just prefer to run 4-6 hour voyages because of the quick turnaround times for quick chron farming, the difference between weighted and balanced is a big deal.  

I'm just saying that I, personally, would like to see the feature implemented.  If I'm in the minority and it's felt that most players would use or benefit from this feature, then so be it.  I'm just casting my vote.

**Comment by IAmPicard** [comment]
This is confusing. You're saying you know the crew selection with non-primary skill sacrifice will result in overall worse voyage duration, but would still want that option?

If you want your voyage to be shorter, you can recall it at any time.

Or is it the case that you're not sure whether the duration is actually worse and want the option just to test out that hypothesis?

**Comment by SodPen1** [comment]
I apologize for the confusion.  Let me back up and try to start at square one to choose my words more accurately.  And again, let my emphasize that the current methodology for calculating the recommended voyage crew is great and will work for most players in most situations the vast majority of the time.  

I've been playing for almost a year, and I am one of those weird players that delights in keeping spreadsheets and whatnot.  When I first started, I strictly ran the balanced approach for my voyages.  As I delved deeper into the wiki and forum posts, I became aware of the WereDragon variant strategy and the NATE variant strategy.  So I began to flip flop between them and keep a record of how they performed with various crew compliments at various voyage lengths.  Then once I succumbed to temptation and started buying dilithium here and there, I began tracking the min/max mathematics of how the three different strategies would affect overall dilithium costs for voyages at different lengths with different goals.  And by that, I mean that sometimes I run voyages with the goal being to maximize chroniton farming over the shortest amount of time.  Sometimes my voyages are for farming crew trainers and replicator rations, again, hoping to achieve the "sweet spot" ratio of voyage length (to include recall time) vs. dilithium cost.  And sometimes my voyages are for farming 4* crew.  

By using the two variant methods, you often times end up freeing 2-4 crew who can be better utilized in running the faction shuttle missions while your voyage is out.  And, according to my tracking, the different goals at different voyage lengths are often substantially more efficient in the big picture than just running the balanced approach.  Of course, as we all know, a lot of the voyage rewards come down to pure RNG, and it's entirely possible that my anecdotal personal experience is atypical.

The TL;DR version is that, according to my tracking and my experience, figuring in diminishing returns from super long (16+ hour) voyages, and figuring in dilithium costs for refills at various time marks, and figuring in the opportunity cost of having crew available for shuttle missions, running voyages using WereDragon and NATE are more desirable than a purely balanced crew compliment.  In fact, for me, it's somewhere in the 20%-25% of the time range.

**Comment by IAmPicard** [comment]
I think that's the piece that is missing for me. I wasn't aware that your crew compliment affects the rewards you receive (as in, can you maximize the chrons you get during your voyage by selecting certain crew). Can you point me to more details on that?

As far as I know the rewards are purely random, and only a function of the voyage duration; hence all the focus on voyage duration.

**Comment by SodPen1** [comment]
Your previous idea is correct... crew compliment does NOT affect rewards directly.  What I'm saying is that, for people who spend dilithium on antimatter refills, there is a definite benefit to using the two alternate strategies in certain situations and not have voyages just be all about getting the absolute longest voyage length possible before running out of antimatter.  

If you look at the table for dilithium cost for refills at various time marks and factor in the diminishing returns for the length of time the voyage is extended vs. the dilithium cost to do so, and what rewards are available to drop only after certain lengths, then I think why I'm requesting this feature becomes apparent.  

For example, let's say my goal was to farm a few 4* crew members (because I'm a completist working on ticking off collections).  Those won't be available until the 6-hour mark / third dilemma has passed.  And looking at my current dilithium reserve, let's say I've got 2,000 dilithium banked.  So I take a quick peek at the table and determine how many hours I want to spend in the "sweet spot" and how much of my dilithium I want to spend on this voyage in total.  We'll say I decide that running a 10-hour voyage (so, four hours with 4* drops being a possibility) is what I'm comfortable with right now as far as spending dilithium.  That's the scenario.

Now, depending on if I use the traditional balanced method, the WereDragon method, or the NATE method, the voyage will fall into three different estimations of length before the first refill.  And this is the key factor, because regardless of compliment, once you start failing every skill check, the amount of time your voyage is extended by a dilithium refill is static and universal among the three.  So determining the length of time to that first refill is the prime concern.  Because that GREATLY affects dilithium expense for subsequent refills because it will set your refill points at different places at different intervals on the chart.  So depending if you are planning on running a 4-hour voyage, a 10-hour voyage, a 16-hour voyage, or a 36-hour voyage, the way you set up that initial crew compliment and how that will play out down the road as far as dilithium expense and what your goal for the voyage is is vitally important.  In pragmatic terms, it can literally mean the difference between wasting $25 real world dollars when you don't have to or not.  And, also, for shorter missions, if you have some legendary crew who can handle it, there's no reason not to have 2-4 unnecessary high-end crew tied up with a voyage if they can be better utilized elsewhere (shuttle missions) while that voyage is out.

This was just a request.  If you don't feel that it's a priority for you or beneficial for most players, I understand.  I won't be offended.

**Comment by paulbilnoski** [comment]
Thanks for the discussion and help understanding different ways to utilize voyages, it has been interesting to me, since I am f2p, to learn about some of these strategies.

At one point you mentioned you would want to run a voyage with less than optimal crew to both hit a voyage duration target and not have particular (you mentioned 2-4) crew tied up so they could be used in shuttles.

What about this process? It seems to be something I would use:
- use the estimator as-is, excluding auto-populated crew currently on shuttles and crew applicable to the current event
- noticing the voyage is too long or particular crew are used (high-base crew you would rather use on shuttles), add those few crew to the exclude list (and remove any auto-populated you want to consider for the voyage) and re-run the estimator
- repeat estimating and excluding additional crew to use elsewhere until a voyage duration with acceptable crew is reached
- initiate the voyage from the tool

**Comment by Gryphonsdream** [comment]
i have to concure with @paulbilnoski with one minor alternate
that might cover the trick thoe teknicaly all ready covered sort of
* calculate to a set number of hours adding in refills as needed with expected Dil cost calculated and giving way a to over run. 
* perhaps a switch to keep your top shuttle crews out of the calculation.
          ** i should note that as you dont know what your shuttles will need untill you open them this gets tricky and can sevearly limit your use list leaving crew just sitting while you are waiting for every hing to finish. example
          ** "sure Mirror Picard just sit around for the next 12 hours because i thought you might be needed for a shuttle. oops my bad, i guess you were not needed after all"
          ** not  very likely senario but i think you get my point.
presently when i see a crew pop up in my voyage list i realy dont want to use, i just add their name to the DoNotUse list and recalculate, ussualy with little change in time calculation (my worst loss was 40 minutes), but other wise i have found it more than amicable.
